package fmt.febulous;

import android.content.Intent;
import android.widget.Toast;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import org.json.JSONException;
import org.json.JSONObject;

import fmt.febulous.helper.BasicFunctions;


public class FCMNotMessaging extends FirebaseMessagingService {

    BasicFunctions basicFunctions;

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {

        basicFunctions = new BasicFunctions(this);

        if (remoteMessage.getData().size() > 0) {

            try {

                JSONObject json = new JSONObject(remoteMessage.getData().toString());
                sendPushNotification(json);

            } catch (Exception e) {

                Toast.makeText(this, "Exception: " + e.getMessage(), Toast.LENGTH_LONG).show() ;

            }
        }
    }


    private void sendPushNotification(JSONObject json) {

        try {

            JSONObject data = json.getJSONObject("data");

            String title = data.getString("title");
            String message = data.getString("message");

            Intent intent = new Intent(getApplicationContext(), Notifications.class);

            basicFunctions.showSmallFCMNotification(title, message, intent);

        } catch (JSONException e) {

            Toast.makeText(this, "Json Exception: " + e.getMessage(), Toast.LENGTH_LONG).show() ;

        } catch (Exception e) {

            Toast.makeText(this, "Exception: " + e.getMessage(), Toast.LENGTH_LONG).show() ;

        }
    }

}